sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/_schema_registry_client.py

```diff
@@ -171,11 +193,22 @@ def get_schema_id(self, schema_group, schema_name, serialization_type, schema_co
         except AttributeError:
             pass
 
-        return self._generated_client.schema.query_id_by_content(
-            group_name=schema_group,
-            schema_name=schema_name,
-            schema_content=schema_content,
-            x_schema_type=serialization_type,
-            cls=_parse_response_schema_id,
-            **kwargs
-        )
+        try:
+            properties = self._description_to_properties[
+                (schema_group, schema_name, serialization_type, schema_content)
+            ]
+            return properties
+        except KeyError:
+            schema_properties = self._generated_client.schema.query_id_by_content(
+                group_name=schema_group,
+                schema_name=schema_name,
+                schema_content=schema_content,
+                x_schema_type=serialization_type,
+                cls=_parse_response_schema_id,
+                **kwargs
+            )
+            self._description_to_properties[
+                (schema_group, schema_name, serialization_type, schema_content)
+            ] = schema_properties
+            self._id_to_schema[schema_properties.schema_id] = Schema(schema_content, schema_properties)
```

Comment:
it is possible that schema_properties.schema_id is already in the `self._id_to_schame` such that we're overwriting the entry here?.

e.g. if a user calls `get_schema` first, then call `get_schema_id`

Reply:
that's a good question, I think it would overwrite the existing schema. We can do something like the following to ensure it's not overwritten:
```suggestion
            cached_schema = self._id_to_schema.get(schema_properties.schema_id)
            if cached_schema:
                self._id_to_schema[schema_properties.schema_id].schema_properties = schema_properties
            else:
                self._id_to_schema[schema_properties.schema_id] = Schema(schema_content, schema_properties)
```
thoughts?

Reply:
I think we only need to `else` part that when it's not cached -- if it's already cached, then why would we need to update it?

Reply:
originally I was thinking that (when the schema is registered through the portal and not with `client.register_schema`), the call to the service to get schema_properties in `get_schema_id` would create a new `SchemaProperties` object that is different from the `SchemaProperties` object in `schema.schema_properties`. I wanted both the `schema_properties` object in the `_description_to_properties` cache and `schema.schema_properties` in `_id_to_schema` cache to point to be the same object.